src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CSharpFormattingPassBase.cs

```diff
@@ -118,7 +119,9 @@ protected async Task<List<TextChange>> AdjustIndentationAsync(FormattingContext
                     continue;
                 }
 
-                sourceMappingIndentations[originalLocation] = indentation;
+                var scopeOwner = syntaxTreeRoot.LocateOwner(new SourceChange(originalLocation, 0, string.Empty));
+                var containingDirective = scopeOwner?.FirstAncestorOrSelf<RazorDirectiveSyntax>();
```

Comment:
Perhaps its possible to remember the last seen `containingDirective`, and only try to locate the owner, and traverse the tree if we've passed it? Or does that break for C# nested inside sections?